pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -0,0 +1,374 @@
+	if err != nil {
+		return err
+	}
+
+	ovfDescriptor, err := archive.ReadOvf("*.ovf")
+	if err != nil {
+		return errors.Errorf("failed to read ovf: %s", err)
+	}
+
+	ovfEnvelope, err := archive.ReadEnvelope(ovfDescriptor)
+	if err != nil {
+		return errors.Errorf("failed to parse ovf: %s", err)
+	}
+
+	// The RHCOS OVA only has one network defined by default
+	// The OVF envelope defines this.  We need a 1:1 mapping
+	// between networks with the OVF and the host
+	if len(ovfEnvelope.Network.Networks) != 1 {
+		return errors.Errorf("Expected the OVA to only have a single network adapter")
+	}
+	// Create mapping between OVF and the network object
+	// found by Name
+	networkMappings := []types.OvfNetworkMapping{{
+		Name:    ovfEnvelope.Network.Networks[0].Name,
+		Network: importOvaParams.Network.Reference(),
+	}}
+	// This is a very minimal spec for importing
+	// an OVF.
+	cisp := types.OvfCreateImportSpecParams{
+		EntityName:     d.Get("name").(string),
+		NetworkMapping: networkMappings,
+	}
+
+	m := ovf.NewManager(client)
+	spec, err := m.CreateImportSpec(ctx,
+		string(ovfDescriptor),
+		importOvaParams.ResourcePool.Reference(),
+		importOvaParams.Datastore.Reference(),
+		cisp)
+
+	if err != nil {
+		return errors.Errorf("failed to create import spec: %s", err)
+	}
+	if spec.Error != nil {
+		return errors.New(spec.Error[0].LocalizedMessage)
+	}
+
+	//Creates a new entity in this resource pool.
+	//See VMware vCenter API documentation: Managed Object - ResourcePool - ImportVApp
+	lease, err := importOvaParams.ResourcePool.ImportVApp(ctx,
+		spec.ImportSpec,
+		importOvaParams.Folder,
+		importOvaParams.Host)
+
+	if err != nil {
+		return errors.Errorf("failed to import vapp: %s", err)
+	}
+
+	info, err := lease.Wait(ctx, spec.FileItem)
+	if err != nil {
+		return errors.Errorf("failed to lease wait: %s", err)
+	}
+
+	u := lease.StartUpdater(ctx, info)
+	defer u.Done()
+
+	for _, i := range info.Items {
+		// upload the vmdk to which ever host that was first
+		// available with the required network and datastore.
+		err = upload(ctx, archive, lease, i)
+		if err != nil {
+			return errors.Errorf("failed to upload: %s", err)
+		}
+	}
+
+	err = lease.Complete(ctx)
+	if err != nil {
+		return errors.Errorf("failed to lease copmlete: %s", err)
+	}
+
+	d.SetId(info.Entity.Value)
+	log.Printf("[DEBUG] %s: ova import complete", d.Get("name").(string))
+
+	return resourceVSpherePrivateImportOvaRead(d, meta)
+}
+
+func resourceVSpherePrivateImportOvaRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*VSphereClient).vimClient.Client
+	moRef := types.ManagedObjectReference{
+		Value: d.Id(),
+		Type:  "VirtualMachine",
+	}
+
+	vm := object.NewVirtualMachine(client, moRef)
+	if vm == nil {
+		return fmt.Errorf("error VirtualMachine not found")
+	}
+
+	return nil
+}
+
+func resourceVSpherePrivateImportOvaUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceVSpherePrivateImportOvaRead(d, meta)
+}
```

Comment:
these functions seem odd.

Reply:
I used the vsphere terraform provider as an example for the read though I greatly simplified from their version.
I wasn't quite sure if update was required, if not we can always remove it.

Reply:
Removed update and set to `ForceNew`.